onebusaway-transit-data-federation/src/main/java/org/onebusaway/transit_data_federation/impl/realtime/gtfs_realtime/GtfsRealtimeTripLibrary.java

```diff
@@ -545,8 +545,8 @@ private int computeArrivalTime(StopTimeEntry stopTime,
       return stopTime.getArrivalTime() + arrival.getDelay();
     if (arrival.hasTime())
       return (int) (arrival.getTime() - serviceDate / 1000);
-    throw new IllegalStateException(
-        "expected arrival delay or time for stopTimeUpdate " + stopTimeUpdate);
+	
+	return -1;
```

Comment:
It doesn't appear that I can comment on Github on unchanged lines of code, so I'll comment here instead - this same conceptual change should be made in `computeDepartureTime()` below - the same problem would occur with an empty departure time.  Also recommend log statement and comment there as well.